posts/openscanhub-prototype/index.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "SAST using OpenScanHub is here!"
+date: 2024-07-30T08:46:18+02:00
+authors: lbarczio
+tags:
+  - copr
+  - srpm
+  - sast
+  - openscanhub
+  - shift left
+---
+
+We are excited to announce a new experimental feature in our service: the integration of Static
+Application Security Testing (SAST) using [OpenScanHub](https://openscanhub.fedoraproject.org/).
+
+OpenScanHub is a service that runs various static analyzers on RPM packages - by default `Cppcheck`,
+`ShellCheck`, the static analyzers embedded in `GCC` and `Clang`, and the `find-unicode-control` tool.
+
+Let's have a look at the details of the prototype!
+
+<!--truncate-->
+
+## Why
+
+This initiative contributes to the _shift left_ effort, aiming to detect and address security
+vulnerabilities earlier in the development process, thus enhancing overall software quality
+and security. By implementing this scanning functionality for `fedora-rawhide`, the current
+development version of Fedora, we aim to catch security issues at the earliest possible stage.
+This is also important as Fedora serves as an upstream platform for downstream distributions
+like RHEL, ensuring that any vulnerabilities are addressed before they propagate to these other systems.
+
+## Functionality
+
+The functionality is designed to automatically
+scan for vulnerabilities and issues in code submitted through pull requests.
+It is controlled using the configuration option
+[`osh_diff_scan_after_copr_build`](/docs/configuration#osh_diff_scan_after_copr_build),
+which is currently set to `true` by default.
+However, even with this option enabled, the differential scanning feature in OpenScanHub
+will only run for users who have also configured Copr builds with the `trigger: commit`
+setting and have matching target branches in their pull request and job configurations.
+This setup is crucial as it allows Packit to access the base build necessary for differential
+scanning, leading to more precise detection of issues introduced by the pull request.
+
+To utilise this feature, you must ensure that Copr builds for commits are configured,
+along with Copr builds for pull requests, both set to run for `fedora-rawhide-x86_64`.
+If you haven't enabled Copr builds for commits yet, you can add the following job
+configuration to the `jobs` section in your Packit configuration:
+
+```yaml
+- job: copr_build
+  trigger: commit
+  branch: main
+  targets:
+    - fedora-rawhide
```

Comment:
Don't we want to suggest stable project?

```suggestion
- job: copr_build
  trigger: commit
  branch: main
  preserve_project: True
  targets:
    - fedora-rawhide
```

Reply:
Oh, I see you are mentioning this later but this can help to avoid possible issues on projects without regular contributions...